### PR TITLE
Merge xenops-cli

### PR DESCRIPTION
**Pull request by robhoes**
This merge brings xenops-cli.git back home. The repo is first added into the directory `xenops-cli/`, while preserving its history (hence the large number of commits under this PR). This was done with the following command:

    git subtree add --prefix xenops-cli git@github.com:xapi-project/xenops-cli master

Then the core files are then moved to `cli/`, updated to fit into the local build, and the remnants of xenops-cli removed.

### CODE REVIEW

**Comment by robhoes** [comment]
It is hard to see that history is preserved in the above patches. I hope the PR merge will do the right thing...

File: cli/dune [hunk 1]
```diff
@@ -25,8 +25,14 @@
  )
 )
 
+(rule
+ (targets xenops-cli.1.gz)
+ (action
+  (run gzip %{dep:xenops-cli.1}))
```
Comment (psafont) [inline cli/dune]: Would gzip have to added to the opam depexts?

Comment (robhoes) [reply cli/dune]: Hmm... there isn't any `depext:` in the opam files of xenopsd... There must be a lot more missing. And I'm not sure about our other repos. Is this something we care to maintain?

Comment (psafont) [reply cli/dune]: If it's not needed to compile the code I don't think anyone will care very much...

**Pull request merged**